File: 1716-calculate-money-in-leetcode-bank/1716-calculate-money-in-leetcode-bank.py
class Solution:
    def totalMoney(self, n: int) -> int:
        balance = 1
        deposits = {0: 1}

        for d in range(1, n):
            depositAmt = 0
            weekDay = d % 7
            if weekDay == 0:
                balance += deposits[0] + 1
                depositAmt = deposits[0] + 1

            else:
                balance += deposits[weekDay - 1] + 1
                depositAmt = deposits[weekDay - 1] + 1

            deposits[weekDay] = depositAmt

        return balance

File: 1716-calculate-money-in-leetcode-bank/1716-calculate-money-in-leetcode-bank.cpp
class Solution {
public:
    int totalMoney(int n) {
        int balance = 0;
        int week = 1;
        while (n > 7) {
            balance += (week * 7) + 21;
            n -= 7;
            ++week;
        }
        int day = week;
        while (n > 0) {
            balance += day;
            ++day;
            --n;
        }
        return balance;
    }
};